Sockets/BackEndMonitoringSockets/src/CAcceptor.cpp

```diff
@@ -1,19 +1,21 @@
 #include "stdafx.h"
 
 #include "CAcceptor.h"
-#include "CServiceConnectionHandler.h"
-#include "EEventType.h"
 #include "CLogger/include/Log.h"
 #include "CEvent.h"
 
 CAcceptor::CAcceptor(bool is_blocked, int socket_timeout, CEvent& event)
 	: m_is_socket_blocked(is_blocked), m_socket_timeout(socket_timeout),
-	m_event(event), m_port(c_not_initialized_port),
-	m_is_acceptor_initialized(false), m_is_time_out(false)
+	m_event(event), m_is_acceptor_initialized(false), m_is_time_out(false)
 { }
 
+CAcceptor::~CAcceptor() noexcept
+{
+	PlatformUtils::CloseSocket(m_p_socket_acceptor->GetSocketFD());
+}
+
 bool CAcceptor::Initialize(const std::string& ip_address,
-	const int listener_port, const int connections)
+                           const int listener_port, const int connections)
```

Comment:
Please add protection from calling this method twice.

Reply:
Added protection in classes CAcceptor, CConnector, CServiceHost, CClientConnectorHost